File: InsertionSort/insertionSort.js
function insertionSort(array){
    for (let i = 1; i < array.length; i++) {
        let key = array[i];
        for (var j = i-1; j >= 0; j--) {
            if(array[j] > key){
                array[j+1] = array[j];

            }
            else{
                break;
            }
        }
        array[j+1] = key; 
    }
    console.log(array);
}



insertionSort([9, 5, 1, 4, 3]);

File: InsertionSort/insertionSort.py
def insertionSort(array: list):
    for i in range(1, len(array)):
        key = array[i]
        j = i - 1
        while(j >= 0 and key < array[j]):
            if array[j] > key:
                array[j+1] = array[j]
            j = j - 1
        array[j+1] = key

    return array


print(insertionSort([1, 3, 5, 0, 4]))   
    
